astropy/io/fits/scripts/fitsheader.py

```diff
@@ -451,6 +458,20 @@ def main(args=None):
     if args.table is None:
         args.table = 'ascii.fixed_width'
 
+    # TODO: This is imho a little nicer than the custom action approach requiring
+    #  the store keyword.
+    #  Approach should be unified to whatever method is chosen
+    if args.sort:
+        args.sort = [key.replace('.', ' ') for key in args.sort]
+        if not args.fitsort:
+            # TODO Only really makes sense for fitsort format. Maybe for standard
+            #  but then it would chug for a while and not start printing
+            #  immediately if you pass many files.
+            #  Maybe this should be an error, but that seems hard
+            #  to do natively with argparse, i.e. exception looks different
+            log.warning('Sorting is only supported in conjunction with -f/--fitsort. '
+                        'Sorting keywords are ignored.')
```

Comment:
I think it's fine to support sorting only with fitsort. And yes it's probably better to raise an exception, or just use `sys.exit` to terminate the program (not sure what's the issue with an exception, but I usually use `sys.exit` in that case).

Reply:
I was trying to find a way to get `argparse` to report the error, like it does for other invalid arguments.
```
usage: fitsheader [-h] [--version] [-e HDU] [-k KEYWORD] [-t [FORMAT]] [-f] [-c] filename [filename ...]
fitsheader: error: the following arguments are required: filename
```
An exception prints the whole stack-trace so it looks quite different to the user. Adopted `sys.exit` method